pulpcore/app/tasks/importer.py

```diff
@@ -58,22 +61,52 @@ def _destination_repo(importer, source_repo_name):
     return Repository.objects.get(name=dest_repo_name)
 
 
+def _impfile_iterator(fd):
+    """
+    Iterate over an import-file returning batches of rows as a json-array-string.
+
+    We use naya.json.stream_array() to get individual rows; once a batch is gathered,
+    we yield the result of json.dumps() for that batch. Repeat until all rows have been
+    called for.
+    """
+    eof = False
+    batch = []
+    rows = stream_array(tokenize(fd))
+    while not eof:
+        try:
+            while len(batch) < IMPORT_BATCH_SIZE:
+                batch.append(next(rows))
+        except StopIteration:
+            eof = True
+        yield json.dumps(batch)
+        batch.clear()
```

Comment:
Rewriting in order to understand again...

```suggestion
    for row in rows:
        batch.append(row)
        if len(batch) >= IMPORT_BATCH_SIZE:
            yield json.dumps(batch)
            batch.clear()
    yield json.dumps(batch)
```

Reply:
Take it if you like, and merge anyway.

Reply:
Working on it, today has been interrupt-driven to an excessive degree, prob won't get this change resubmitted till tomorrow.